### PR TITLE
Improve URL creation and deletion UX / UI output

### PR DESCRIPTION
**Pull request by cdrage**
This PR:
 - Improves the UI / UX of the URL output
 - Separates Validation / Creation / Updating of components in the CLI
 output

Closes https://github.com/openshift/odo/issues/1648

### CODE REVIEW

**Comment by cdrage** [comment]
Preview here:

[asciicast](https://asciinema.org/a/Szh9pBMWL5Rbg4gGVC0MvLHdi)

**Comment by cdrage** [comment]
Sorry for not having tests yet @amitkrout and @girishramnani I'll have one for checkIfURLChangesWillBeMade soon :)

**Comment by cdrage** [comment]
I've updated the UI in this PR too and would like some feedback @girishramnani @kadel @surajnarwade 

[asciicast](https://asciinema.org/a/1RC6N4qJMrVnJqSKv2cbvgsyI)

I have removed the "deploying to component" messages and made this more seamless / minimal.

**Comment by kadel** [comment]
/retest
```
[odo]  ✗  Unable to connect to OpenShift cluster, is it down?
```

**Comment by girishramnani** [comment]
/approve
once tests are present will `lgtm` @cdrage ?

**Comment by kadel** [comment]
> /approve
> once tests are present will `lgtm` @cdrage ?

What kind of test would you like to see here?

**Comment by girishramnani** [comment]
`Write test for checkIfURLChangesWillBeMade` @kadel

File: pkg/component/component.go [hunk 1]
```diff
@@ -1372,3 +1365,21 @@ func getStorageFromConfig(localConfig *config.LocalConfigInfo) storage.StorageLi
 	}
 	return storageList
 }
+
+// checkIfURLChangesWillBeMade checks to see if there are going to be any changes
+// to the URLs when deploying and returns a true / false
+func checkIfURLChangesWillBeMade(client *occlient.Client, componentConfig config.LocalConfigInfo) (bool, error) {
```
Comment (cdrage) [inline pkg/component/component.go]: I am assuming you want a test for this one here @girishramnani

Comment (girishramnani) [reply pkg/component/component.go]: Yes

Comment (kadel) [reply pkg/component/component.go]: This is not exported function. Not sure if it is that important to have this covered.

`ApplyConfig` that is calling this should be definitely covered and have test cases for this. (Doesn't have to be part of this PR)

@girishramnani @cdrage The decision is up to you.

**Comment by cdrage** [comment]
Ready for another round of review @kadel @girishramnani @surajnarwade @amitkrout @mik-dass @dharmit 

Preview of the changes:

[asciicast](https://asciinema.org/a/ioZFJ7w5R70vNeXFCnGmSKXEg)

**Comment by surajnarwade** [comment]
```
$ odo create nodejs
 ✓  Validating component
Please use `odo push` command to create the component with source deployed
```

```
$ odo push
Validation
 ✓  Validating component
 ✓  Checking component

Configuration changes
 ✓  Initializing component
 ✓  Creating component
 ✓  Applying configuration

Pushing to component nodejs-xnlo of type local
 ✓  Waiting for component to start
 ✓  Copying files to component
 ✓  Building component
 ✓  Changes successfully pushed to component
```

```
 $ odo url create --port 8080
 ✓  URL nodejs-xnlo-8080 created

Run `odo push` to apply URL: nodejs-xnlo-8080
```

```
$ odo push
Validation
 ✓  Validating component
 ✓  Checking component

Configuration changes
 ✓  Applying configuration

Applying URL changes
 ✓  URL nodejs-xnlo-8080: http://nodejs-xnlo-8080-app-myproject.192.168.42.120.nip.io created

Pushing to component nodejs-xnlo of type local
 ✓  Waiting for component to start
 ✓  Copying files to component
 ✓  Building component
 ✓  Changes successfully pushed to component
```


@cdrage tested locally, WFM :+1: 
went through code, LGTM :tada:

**Pull request merged**